drivers/spi/spi_nrfx_spis.c

```diff
@@ -124,6 +131,22 @@ static int prepare_for_transfer(const struct device *dev,
 		return -EINVAL;
 	}
 
+	err = dmm_buffer_out_prepare(dev_config->mem_reg, tx_buf, tx_buf_len, (void **)&tx_buf);
+	if (err != 0) {
+		LOG_INF("tx alloc err=%d", err);
+		return err;
+	}
+
+	err = dmm_buffer_in_prepare(dev_config->mem_reg, rx_buf, rx_buf_len, (void **)&rx_buf);
+	if (err != 0) {
+		LOG_INF("tx alloc err=%d", err);
```

Comment:
buffers needs to be released in case of error. Ref: https://github.com/zephyrproject-rtos/zephyr/pull/71278#discussion_r1638051341